pyuavcan/transport/serial/_frame.py

```diff
@@ -105,23 +105,37 @@ def compile_into(self, out_buffer: bytearray) -> memoryview:
 
         payload_crc_bytes = pyuavcan.transport.commons.crc.CRC32C.new(self.payload).value_as_bytes
 
-        escapees = self.FRAME_DELIMITER_BYTE, self.ESCAPE_PREFIX_BYTE
         out_buffer[0] = self.FRAME_DELIMITER_BYTE
         next_byte_index = 1
-        for nb in itertools.chain(header, self.payload, payload_crc_bytes):
-            if nb in escapees:
-                out_buffer[next_byte_index] = self.ESCAPE_PREFIX_BYTE
-                next_byte_index += 1
-                nb ^= 0xFF
-            out_buffer[next_byte_index] = nb
-            next_byte_index += 1
+
+        packet_bytes = header + self.payload + payload_crc_bytes
+        encoded_image = cobs.encode(packet_bytes)
+        # place in the buffer and update next_byte_index:
+        out_buffer[next_byte_index:next_byte_index] = encoded_image
+        next_byte_index += len(encoded_image)
 
         out_buffer[next_byte_index] = self.FRAME_DELIMITER_BYTE
         next_byte_index += 1
 
         assert (next_byte_index - 2) >= (len(header) + len(self.payload) + len(payload_crc_bytes))
         return memoryview(out_buffer)[:next_byte_index]
 
+    @staticmethod
+    def calc_cobs_size(payload_size_bytes: int) -> int:
+        return int(math.ceil(payload_size_bytes * 255.0 / 254.0))
```

Comment:
It is better to avoid floating point when dealing with discrete entities like bytes for simplicity and to provide cleaner implementation examples for people who will be re-implementing this logic for embedded systems.

```suggestion
        return (payload_size_bytes * 255 + 253) // 254
```

This change is equivalent which can be verified:

```python
>>> set(int(math.ceil(payload_size_bytes * 255.0 / 254.0)) - ((payload_size_bytes * 255 + 253) // 254) for payload_size_bytes in range(10000000))
{0}
```

The `math` import is no longer needed.